src/System.CommandLine.Suggest/FileSuggestionRegistration.cs

```diff
@@ -52,9 +53,18 @@ public Registration FindRegistration(FileInfo soughtExecutable)
                 return null;
             }
 
-            var completionTarget =
-                File.ReadAllLines(_registrationConfigurationFilePath).LastOrDefault(line =>
-                                                                                        line.StartsWith(soughtExecutable.FullName, StringComparison.OrdinalIgnoreCase));
+            string completionTarget = null;
+            using (var sr = new StreamReader(_registrationConfigurationFilePath, Encoding.UTF8))
+            {
+                string? line;
+                while ((line = sr.ReadLine()) != null)
+                {
+                    if (line.StartsWith(soughtExecutable.FullName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        completionTarget = line;
+                    }
+                }
+            }
 
             if (completionTarget == null)
```

Comment:
Maybe change to `if (completionTarget is not null)`?

Reply:
Don't you mean `if (completionTarget is null)`?

Reply:
Oops. Yeah.